packages/nimble-components/src/select/index.ts

```diff
@@ -969,39 +1017,131 @@ export class Select
         this.updateListboxMaxHeightCssVariable();
     }
 
+    private updateAdjacentSeparatorState(
```

Comment:
I spent a long time staring at this function trying to succinctly describe the rules for when separators should be visible. It seems like it should be simpler to understand.

I think it boils down to:
1. if I'm a group, show the bottom separator if there is a nextElement
2. if I'm a group, show the top separator if the previous element is an option

Among the things that would be good to clean up:
1. why do we set `element.bottomSeparatorVisible` regardless of `isOptionOrGroupVisible(element)` but only set `element.topSeparatorVisible` if `element` is visible?
2. instead of having one method encode the logic for both this element and next/previous and for both groups and options, have a lower-level method that sets the state for a given group and call that method on the 3 relevant groups (this, previous, next)
3. find a way to make the check-if-I'm-a-group-then-set-state pattern less verbose (1 line instead of 3). Possibly another helper function or maybe this will go away after you do 2.

Reply:
I've now refactored this, and hopefully it is more clear.